script/openqa-powermanagement.py

```diff
@@ -0,0 +1,123 @@
+  
+# Create list of WORKER_CLASS needed
+for job in scheduled_list_data['data']:
+  response = requests.get(openqa_server + '/api/v1/jobs/' + str(job['id']))
+  job_data = json.loads(response.content)
+  jobs_worker_classes.append(job_data['job']['settings']['WORKER_CLASS'])
+
+jobs_worker_classes = sorted(set(jobs_worker_classes))
+print("Found " + str(len(jobs_worker_classes)) + " different WORKER_CLASS in scheduled jobs: " + str(jobs_worker_classes))
+
+
+
+# Workers
+workers_list_file = requests.get(openqa_server + '/api/v1/workers').content
+workers_list_data = json.loads(workers_list_file)
+
+# Create list of hosts which may need to powered up/down
+for worker in workers_list_data['workers']:
+  if worker['status'] in ['idle']:
+    machine_list_idle.append(worker['host'])
+  elif worker['status'] in ['dead']: # Looks like 'dead' means 'offline'
+    machine_list_offline.append(worker['host'])
+  else: # worker['status'] in ['running', 'broken']: # Looks like  'running' means 'working'
+    machine_list_busy.append(worker['host'])
+
+# Clean-up the lists
+machine_list_idle = sorted(set(machine_list_idle))
+machine_list_offline = sorted(set(machine_list_offline))
+machine_list_busy = sorted(set(machine_list_busy))
+
+# Remove the machine from idle/offline lists if at least 1 worker is busy
+for machine in machine_list_busy:
+  if machine in machine_list_idle:
+    machine_list_idle.remove(machine)
+  if machine in machine_list_offline:
+    machine_list_offline.remove(machine)
+# Remove the machine from offline list if at least 1 worker is idle
+for machine in machine_list_idle:
+  if machine in machine_list_offline:
+    machine_list_offline.remove(machine)
+
+# Print an overview
+print(str(len(machine_list_idle)) + " workers listed fully idle: " + str(machine_list_idle))
+print(str(len(machine_list_offline)) + " workers listed offline/dead: " + str(machine_list_offline))
+print(str(len(machine_list_busy)) + " workers listed busy: " + str(machine_list_busy))
+
+# Get WORKER_CLASS for each workers of each machines (idle and offline) and compare to WORKER_CLASS required by scheduled/blocked jobs
+for worker in workers_list_data['workers']:
+  if worker['host'] in machine_list_offline:
+    for classes in jobs_worker_classes:
+      if set(classes.split(',')).issubset(worker['properties']['WORKER_CLASS'].split(',')):
+        machines_to_power_on.append(worker['host'])
+   
+  if worker['host'] in machine_list_idle:
+    if worker['properties']['WORKER_CLASS'] in jobs_worker_classes:
+      # Warning: scheduled (blocked?) job could be run on idle machine!
+      print("Warning: scheduled (blocked?) job could be run on idle machine!")
+
+# Power on machines which can run scheduled jobs
+for machine in sorted(set(machines_to_power_on)):
+  if 'PowerManagement' in config and config['PowerManagement'].get(machine + "_POWER_ON"):
```

Comment:
For the sake of consistency within our INI files:
```suggestion
  if 'power_management' in config and config['power_management'].get(machine + "_POWER_ON"):
```

Reply:
Ok